pdns/dnsproxy.cc

```diff
@@ -250,10 +272,14 @@ void DNSProxy::mainloop()
         memset(&msgh, 0, sizeof(struct msghdr));
         string reply; // needs to be alive at time of sendmsg!
         MOADNSParser mdp(false, p.getString());
-        //	  cerr<<"Got completion, "<<mdp.d_answers.size()<<" answers, rcode: "<<mdp.d_header.rcode<<endl;
+        if (p.d_eso.scope.isValid()){
```

Comment:
I believe we need to check that the received source matches the one we sent, otherwise we might reply with garbage. Thinking about it, perhaps it would be better to only provide read-only access to `DNSPacket::d_eso` and to provide a `DNSPacket` method to request a cap on the ECS scope instead?

Reply:
Indeed, the current code allows forwarding a response that does not match what was asked. I'll push a (failing) test for that.

When there is a mismatch, what do you think we should do? Drop ECS from the response? Just apply the scope?

Reply:
(just apply the scope)

Reply:
Agreed, applying the received scope to the initial ECS value should be fine.